src/main/java/seedu/address/MainApp.java

```diff
@@ -75,7 +76,7 @@ public void init() throws Exception {
 
         ModelManager modelManager = (ModelManager) model;
         UiManager uiManager = (UiManager) ui;
-        modelManager.addUiUpdateListener(uiManager);
+        modelManager.addUiUpdateListener((UiUpdateListener) uiManager);
```

Comment:
Careful with this line! The whole downcasting to (UiUpdateListener) has given me some problems with checkstyle in previous commits